#120 tabify

joshxg opened the pull request:
Fixes #120 

I guess I'm in the mood to be doing things which look visually appealing! I guess it makes me feel like more progress is happening 🤣 

![image](https://user-images.githubusercontent.com/35858975/133689303-7cf36638-ca1b-4685-91ed-09c2300cb8b6.png)


- Future thoughts are to take this and put it into a `DetailViewToolbar` would be good which can have the drop down and filters and stuff. Similar to a `ListViewToolbar` just with tabs, I guess.
- There is some enhancements in `@mui/lab` for tabs, now - 🙄  - primarily, it is an addition of a `TabContext`, and there own `TabPanel`, where the `TabPanel` will look up the `currentTab` in the `Context`. It will also add better aria props. Mainly, each `tabpanel` role should have a `aria-controls` to the `tablist` role. I wasn't sure if I should add the `@mui/lab` package or not, so looking to get your opinion on if I should add the package and use the components, or add the aria stuff myself.
- The tests are a bit flakey - they'd be more useful if there were aria labels indicating a link between the tab pressed and the content. I didn't want to go to far with testing the actual tabs, though, as mui already have a bunch

joshxg commented on packages/common/src/intl/locales/ar.json:
```diff
@@ -47,7 +47,12 @@
+  "label.item": "Item",
+  "label.batch": "Batch",
+  "label.price": "Price",
+  "label.log": "Log",
```
Didn't know what to key these as. Was also thinking these are very generic terms so `generic.` came to mind.. 🤔

joshxg replied:
Yup. There's also something there with sharing localisation files with different projects

joshxg commented on packages/common/src/ui/components/navigation/Tabs/Tab.tsx:
```diff
@@ -0,0 +1,9 @@
+import { styled } from '@material-ui/system';
+
+export const Tab = styled(MuiTab)({
+  color: '#8f90a6',
```
ugh theeres a couple colors I need to sort out in here

joshxg commented on packages/common/src/ui/components/navigation/Tabs/TabPanel.tsx:
```diff
@@ -0,0 +1,24 @@
+        </Box>
+      )}
+    </>
+  );
```
spreading the box props in this one and continuing to be inconsistent. Thought it would be annoying to not be able to add margins if you needed to add margins to your page etc..

joshxg commented on packages/common/src/ui/components/navigation/Tabs/Tabs.tsx:
```diff
@@ -0,0 +1,10 @@
+
+export const Tabs = styled(MuiTabs)({
+  minHeight: '42px',
+  maxHeight: '42px',
```
trying to reduce the gap between the indicator highlight and the text - couldn't find another way!

joshxg commented on packages/transactions/src/OutboundShipment/DetailView/DetailView.test.tsx:
```diff
@@ -1,23 +1,85 @@
 import { Route } from 'react-router';
 import { render, waitFor } from '@testing-library/react';
-
+import AppBar from '../../../../host/src/AppBar';
```
Ew!

I had a lot of thoughts about this..

- `TestingAppBar` .. ?
- Thoughts around component libraries..
- If we want a consistent look and feel to web apps, we likely want the `AppBar` in `common`, in at least, some form.
- We could go further and have an `AppLayout` component, with a drawer and `AppBar` (And RHS Drawer..?)
- Since we the components are pretty tightly coupled to the translation module.. and also there are a bunch of other providers that we're coupled to right now, we might also want them in the `AppLayout`..
- That basically the `Host` minus the routes.
- So `AppLayout` could be a good base for any web app, `Host` uses `AppLayout` and then we can also use `AppLayout` in tests..

and then I thought I needed to back away and have you bring me back down to earth so I left it

mark-prins replied:
It does hollow out the Host and moves a bunch of things into the common app which aren't reused elsewhere.
Aside from testing, I'm not seeing why there would be a need to have multiple AppBar per site or LH drawer etc.

unless, we are thinking to take the whole common bundle and share that across sites / apps. now that would be awesome! I'm thinking then we could have the health supply hub ( and IRIMS, universal codes..?? ) share design and components with open mSupply.

there is real potential there.. though to share the common we'd have to have it in a package, so there's a bit of work later on when we're ready for that too.

joshxg replied:
Yup, having common become a shared package for all web apps is what I was thinking. I wasn't sure about previous projects, but definitely any future ones.

Yeah, having it in a package would be annoying.

`@openmsupply-client` -> `@openmsupply-web`  🤔

mark-prins replied:
yes!! the rename. I like `-web` much better

mark-prins commented on packages/transactions/src/OutboundShipment/DetailView/DetailView.tsx:
```diff
@@ -150,23 +155,53 @@ export const OutboundShipmentDetailView: FC = () => {
+      </AppBarContentPortal>
+
+      <TabPanel tab={0} currentTab={currentTab}>
+        <div>
```
will need to lift the tab contents to their own components soon, this file is going to get huge

joshxg replied:
For sure!

mark-prins reviewed (approved):
👍  looks fantastic!

In the UI meeting yesterday we discussed the tabs for this screen. Hoping to drop `Item` and `Batch` as they are different views of the same info that it is in `General`. Might try and push the `Log` into the detail panel. The `Price` is actually from supplier invoices and doesn't apply to the outbound shipment. There should be the 'Transport' one though 🙄 

So, sorry. Didn't update the issue in time. Would you mind updating just the tabs for now - drop item & batch and rename price to Transport?

I think go with the experimental version of the tab too 🤷‍♂️

joshxg commented on packages/transactions/src/OutboundShipment/DetailView/DetailView.test.tsx:
```diff
@@ -1,23 +1,85 @@
+    expect(itemPanel).toBeInTheDocument();
+  });
+
+  it('Renders the item summary content once the item summary tab has been pressed', async () => {
```
Yeah - tests improved now by using the lab with the accessibility roles added, making it a lot easier to find stuff b072fb8

joshxg commented on packages/common/src/ui/components/navigation/Tabs/Tabs.stories.tsx:
```diff
@@ -0,0 +1,106 @@
+        <Tab value={3} label="Log" />
+        <Tab value={4} label="Transport" />
+      </Tabs>
+      <TabPanel
```
Hehehe I couldn't think of anything else easy to put in the tabs! Have renamed!

joshxg commented:
- Added `@material-ui/lab` : Technically the latest is `@mui/lab`, but that has a dep on `@mui/material` (previously `@material-ui/core`). so I added the 'older' version for now, and can upgrade `@mui/lab` and `@mui/material` at once..
- Just the two tabs. I also thought this was a bit weird and thought that we might be able to do some cool groupings in the table by batch/item etc instead heh
![image](https://user-images.githubusercontent.com/35858975/133866919-2367a68d-d163-48cc-9223-44c71e473c64.png)

mark-prins commented on packages/transactions/src/OutboundShipment/DetailView/DetailView.tsx:
```diff
@@ -150,23 +157,46 @@ export const OutboundShipmentDetailView: FC = () => {
+          </TabPanel>
+
+          <TabPanel value="transport">
+            <span>Price details coming soon..</span>
```
If you spot this before committing.. transport details?

mark-prins reviewed (approved):
that is very nice! 
agree about not moving to `@mui` as part of this, one step at a time :)

joshxg commented:
Oops! I will change the description in a later PR - going to merge this to try and fix the diffs in the other prs 😁
